javascript/computer_science/project_hash_map.md

```diff
@@ -18,17 +18,21 @@ if (index < 0 || index >= buckets.length) {
 
 <div class="lesson-content__panel" markdown="1">
 
-  1. Start by creating a `HashMap` class or factory function. It's up to you which you want to use. Then proceed to create the following methods:
+  Start by creating a `HashMap` class or factory function. It's up to you which you want to use. Then proceed to create the following methods:
 
-  1. `hash(key)` takes a key and produces a hash code with it. We did implement a fairly good `hash` function in the previous lesson. You are free to use that, or if you wish, you can conduct your own research. Beware, this is a deep deep rabbit hole.
+  1. `hash(key)` takes a key and produces a hash code with it. We did implement a fairly good `hash` function in the previous lesson, see below. You are free to use that, or if you wish, you can conduct your own research. Beware, this is a deep deep rabbit hole. Also, there is one edge case with long keys that was not taken into consideration in the function or rather how we applied a modulo `%` operator. JavaScript is unable to hold large numbers precisely. At some point, calculations are going to be inacccurate, which significantly increases the chances of collisions. There are a few ways how we could handle it but we recommend that you apply the modulo operator on *each iteration* instead of outside the loop at the end. In that case, we prevent the output from becoming larger than our bucket's length.
 
-      <div class="lesson-note lesson-note--tip" markdown="1">
-        Hash maps could accommodate various data types for keys like numbers, strings, objects. But for this project, only handle keys of type strings.
-      </div>
+  You might find yourself confusing keys with hash codes while accessing key-value pairs later. We would like to stress that the key is what your `hash` function will take as an input. In a way, we could say that the key is important for us only inside the `hash` function. But we never access a bucket directly with the key. Instead we do so with the hash code.
```

Comment:
This paragraph (starting with "You might find yourself") is causing some formatting issues within the assignment's list. You can see the preview for this content [here](https://www.theodinproject.com/lessons/preview?uuid=e2b838b3-10eb-4ec3-b2b9-f84f9e80d5f6). Notice how the list flow is interrupted (there are multiple items labeled with 1), and how the indent is incorrect.

When breaking up list items into multiple lines like you have here, you want to mind [these rules](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#multi-line-list-items) in the LAYOUT_STYLE_GUIDE to make sure it formats correctly.

I'll call out a few other places this is happening in this PR as well.